### PR TITLE
Check that stdout is connected to tty before applying color coding

### PR DESCRIPTION
**Pull request by joakim-hove**
Upstream: https://github.com/OPM/opm-common/pull/2490

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/simulators/utils/readDeck.cpp [hunk 1]
```diff
@@ -151,7 +152,10 @@ FileOutputMode setupLogging(int mpi_rank_, const std::string& deck_filename, con
     if (mpi_rank_ == 0) {
         std::shared_ptr<Opm::StreamLog> streamLog = std::make_shared<Opm::StreamLog>(std::cout, Opm::Log::StdoutMessageTypes);
         Opm::OpmLog::addBackend(stdout_log_id, streamLog);
-        streamLog->setMessageFormatter(std::make_shared<Opm::SimpleMessageFormatter>(true));
+
+        // Check if stdout is connected to a terminal or redirected to a file
+        bool use_color_coding = isatty(fileno(stdout));
```
Comment (bska) [inline opm/simulators/utils/readDeck.cpp:157]: You need `<stdio.h>` for `stdout`.

**Comment by joakim-hove** [comment]
I'll expose `stdoutIsTerminal()` from OpmLog.

**Comment by bska** [comment]
> I'll expose `stdoutIsTerminal()` from OpmLog.

Very good.  I think that's the best option here.

**Review by bska — APPROVED** [review]
Thanks for the updates.  This looks good to me.

**Comment by joakim-hove** [comment]
jenkins build this opm-common=2490 please

**Comment by bska** [comment]
Build check is green.  I'll merge into master.

**Pull request merged**